packages/datadog-plugin-winston/test/index.spec.js

```diff
@@ -72,76 +75,150 @@ describe('Plugin', () => {
-          tracer.scope().activate(span, () => {
-            logger.info('message')
+            tracer.scope().activate(span, () => {
+              logger.info('message')
+
+              if (semver.intersects(version, '>=3')) {
+                expect(transport.log).to.have.been.calledWithMatch(meta)
+              } else {
+                expect(transport.log).to.have.been.calledWithMatch('info', 'message', meta)
+              }
+            })
+          })
 
+          if (semver.intersects(version, '>=3')) {
+            it('should add the trace identifiers when streaming', () => {
+              const logger = winston.createLogger({
+                transports: [transport]
+              })
+
+              tracer.scope().activate(span, () => {
+                logger.write({
+                  level: 'info',
+                  message: 'message'
+                })
+
+                expect(transport.log).to.have.been.calledWithMatch({
+                  dd: {
+                    trace_id: span.context().toTraceId(),
+                    span_id: span.context().toSpanId()
+                  }
+                })
+              })
+            })
+          }
+        })
+
+        describe('with splat formatting', () => {
+          beforeEach(() => {
             if (semver.intersects(version, '>=3')) {
-              expect(transport.log).to.have.been.calledWithMatch(meta)
+              const splatConfiguration = {
+                format: winston.format.combine(...[winston.format.splat(), winston.format.json()])
+              }
+              setup(version, splatConfiguration)
             } else {
-              expect(transport.log).to.have.been.calledWithMatch('info', 'message', meta)
+              setup(version)
             }
           })
-        })
 
-        if (semver.intersects(version, '>=3')) {
-          it('should add the trace identifiers when streaming', () => {
-            const logger = winston.createLogger({
-              transports: [transport]
+          it('should ensure interpolated logs are persisted', () => {
+            const base = 'test'
+            const extra = 'message'
+            const interpolatedLog = base + ` ${extra}`
+            const splatFormmatedLog = base + ' %s'
+
+            const meta = {
+              dd: {
+                trace_id: span.context().toTraceId(),
+                span_id: span.context().toSpanId()
+              }
+            }
+
+            tracer.scope().activate(span, () => {
+              winston.info(splatFormmatedLog, extra)
+
+              if (semver.intersects(version, '>=3')) {
+                meta['message'] = interpolatedLog
+
+                expect(transport.log).to.have.been.calledWithMatch(meta)
+              } else {
+                expect(transport.log).to.have.been.calledWithMatch('info', interpolatedLog, meta)
+              }
             })
+          })
+
+          it('should ensure symbol meta properties are persisted', () => {
```

Comment:
This test is already covered in the the log util tests. It's also an implementation detail that Winston uses symbols for splat formatting, so the above test should be enough.

Reply:
👍 updated